src/app/shared/components/add-match/select-players/select-players.component.ts

```diff
@@ -19,4 +21,98 @@ export class SelectPlayersComponent {
   matchForm: FormGroup;
 
   teams: number[] = [0, 1];
+
+  allPlayers: IUser[] = [];
+
+  playerLists: IUser[][] = [[], [], [], []];
+
+  selectedPlayers: string[] = ['', '', '', ''];
+
+  touched: boolean[] = [false, false, false, false];
+
+  constructor(userService: UsersService, matchesService: MatchesService) {
+    userService.users$.subscribe((data: IUser[]) => {
```

Comment:
Mostly we use ngOnInit for all the initialization/declaration and avoid stuff to work in the constructor. The constructor should only be used to initialize class members but shouldn't do actual "work".

So you should use constructor() to setup Dependency Injection and not much else. ngOnInit() is better place to "start" - it's where/when components' bindings are resolved.